src/ngraph/runtime/hybrid/hybrid_backend.cpp

```diff
@@ -0,0 +1,226 @@
+    {
+        throw std::invalid_argument("read_vector type must match Tensor type");
+    }
+    size_t element_count = ngraph::shape_size(tv->get_shape());
+    size_t size = element_count * sizeof(T);
+    std::vector<T> rc(element_count);
+    tv->read(rc.data(), 0, size);
+    return rc;
+}
+
+shared_ptr<runtime::Backend> runtime::hybrid::HYBRIDBackend::get_cached_backend(Placement placement)
+{
+    if (m_cached_backends.find(placement) == m_cached_backends.end())
+    {
+        m_cached_backends[placement] = runtime::Backend::create(placement_to_string(placement));
+    }
+    return m_cached_backends.at(placement);
+}
+
+shared_ptr<runtime::Tensor> runtime::hybrid::HYBRIDBackend::create_tensor(const element::Type& type,
+                                                                          const Shape& shape)
+{
+    return make_shared<runtime::HostTensor>(type, shape, "external");
+}
+
+shared_ptr<runtime::Tensor> runtime::hybrid::HYBRIDBackend::create_tensor(const element::Type& type,
+                                                                          const Shape& shape,
+                                                                          void* memory_pointer)
+{
+    return make_shared<runtime::HostTensor>(type, shape, memory_pointer, "external");
+}
+
+bool runtime::hybrid::HYBRIDBackend::compile(shared_ptr<Function> function)
+{
+    NGRAPH_INFO << "hybrid compile -Begin ";
+    if (m_function_map.find(function) == m_function_map.end())
+    {
+        // Clone function
+        FunctionInstance instance;
+        instance.m_function = clone_function(*function);
+
+        pass::Manager pass_manager;
+
+        // fall back to CPU as the base transformer 
+        // pass_manager.register_pass<pass::AssignPlacement>(
+        //     runtime::interpreter::default_placement_policy);
+
+        // fall back to Interpreter as the base transformer 
+        pass_manager.register_pass<pass::AssignPlacement>(runtime::cpu::default_placement_policy);
+        
+        pass_manager.run_passes(instance.m_function);
+
+        NGRAPH_INFO << "hybrid compile -begin split  ";
+        // Split function to sub_functions
+        tie(instance.m_sub_functions, instance.m_map_parameter_to_result) =
+            split_function_by_placement(instance.m_function);
+        NGRAPH_INFO << "hybrid compile -End split  ";
+
+        m_function_map.insert({function, instance});
+        NGRAPH_INFO << "hybrid compile -map incertion successful";
+
+        // Compile subfunctions in corresponding backends
+        for (shared_ptr<Function>& sub_function : instance.m_sub_functions)
+        {
+            Placement placement = get_colocated_function_placement(sub_function);
+            auto backend = get_cached_backend(placement);
+            backend->compile(sub_function);
+        }
+    }
+    NGRAPH_INFO << "hybrid compile -End ";
+    return true;
+}
+
+bool runtime::hybrid::HYBRIDBackend::call(shared_ptr<Function> function,
```

Comment:
This entire function is too inefficient. Too many new objects created. We have to make this work with minimal new objects and copies.